### PR TITLE
Ranking saving db

### LINKED ISSUE #41: Add scoring system for new discoverd peaks

### PR DESCRIPTION
**Pull request by bgabriel998**
## Added Scoring system for user
- Create Database containing all countries and their highest peaks. The method is communicating with the database through a cache to speed up the comparisons between the discovered peak and the highest peak from the country.
- Add method to determine the country by giving lat. long. inputs.
 
- Add methods to update user's current score. 
- Add method definitions (empty) to prepare user database implementation
closes #41

### CODE REVIEW

File: app/src/androidTest/java/com/github/giommok/softwaredevproject/AccountTest.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 @RunWith(AndroidJUnit4.class)
 public class AccountTest {
 
+
```
Comment (giommok) [inline app/src/androidTest/java/com/github/giommok/softwaredevproject/AccountTest.java:20]: Why this new empty line?

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 2]
```diff
@@ -52,6 +52,12 @@
     private static final int GIVEN_QUERY_MAX_RESULT = 30;
     private static final int GIVEN_QUERY_TIMEOUT = 10;
 
+    private static final double MOCK_LOCATION_LAT_LAUSANNE = 46.519251915333676;
+    private static final double MOCK_LOCATION_LON_LAUSANNE = 6.558563221333525;
+    private static final double MOCK_LOCATION_ALT_LAUSANNE = 220;
+
+    private static final int MS_TO_SEC = 1000;
```
Comment (magangloff) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:59]: Duplicate of MILLI_SEC_TO_SEC

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 3]
```diff
@@ -64,20 +64,19 @@
...
-                queryTimeS = ((double)System.currentTimeMillis() - startTimeMs)/ MILLI_SEC_TO_SEC;
-            }
-        }.execute();
-        try {
-            Thread.sleep(4000);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-            fail("testResultType failed");
+        final boolean[] dataAcquired = {false};
+        while(!dataAcquired[0]) {
```
[the remark this answers is not in the document]
Comment (bgabriel998) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java]: Resolved in last commit

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 4]
```diff
@@ -112,7 +115,7 @@ public void testResultType(){
      */
     @Test
     public void testResultNameNonNull(){
-        if (resultPOI == null) fail("testResultNameNonNull failed");
+        if (resultPOI == null) fail("testResultNameNonNull failed. Acquired POI List is empty...");
         for(POI point : resultPOI){
             assertNotEquals(point.mType,isEmptyOrNullString());
```
Comment (ravifrancesco) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:120]: Could use assertTrue

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java [hunk 5]
```diff
@@ -138,7 +141,7 @@ public void testResultHeightNonNull(){
      */
     @Test
     public void testResultListNotExceedLimit(){
-        if (resultPOI == null) fail("testResultListNotExceedLimit failed");
+        if (resultPOI == null) fail("testResultListNotExceedLimit failed. Acquired POI List is empty...");
         for(POI point : resultPOI){
             assertThat(resultPOI.size(),lessThanOrEqualTo(DEFAULT_QUERY_MAX_RESULT));
```
Comment (ravifrancesco) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:146]: Could use assert LessEqual

**Review by ravifrancesco — COMMENTED** [review]
Looks good to me, left a couple of suggestions. Also I am getting some warnings on duplicate blocks of code, and what are the two TODOs?

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java (hunk 4 above)
Comment (bgabriel998) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:120]: Convert it to `assertNotNull`

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java (hunk 5 above)
Comment (bgabriel998) [reply app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/GeonamesHandlerTest.java:146]: Convert it to `assertNotNull`

**Review by magangloff — APPROVED** [review]
Seems all good now for me

**Review by ravifrancesco — APPROVED** [review]

**Pull request merged**